src/DonationForms/Actions/ConvertDonationFormBlocksToFieldsApi.php

```diff
@@ -247,6 +252,51 @@ protected function createNodeFromDonorNameBlock(BlockModel $block): Node
         });
     }
 
+    /**
+     * @unreleased
+     */
+    protected function createNodeFromBillingAddressBlock(BlockModel $block): Node
+    {
+        $countryList = [];
+        foreach (give_get_country_list() as $value => $label) {
+            $countryList[] = [$value, $label];
+        }
+
+        return BillingAddress::make('billingAddress')->tap(function ($group) use ($block, $countryList) {
+            $group->getNodeByName('country')
+                ->label($block->getAttribute('countryLabel'))
+                ->options(...$countryList)
+                ->rules('required');
+
+            $group->getNodeByName('address1')
+                ->label($block->getAttribute('address1Label'))
+                ->placeholder($block->getAttribute('address1Placeholder'))
+                ->rules('required', 'max:255');
+
+            $group->getNodeByName('address2')
+                ->label($block->getAttribute('address2Label'))
+                ->placeholder($block->getAttribute('address2Placeholder'))
+                ->required($block->getAttribute('requireAddress2'))
+                ->rules('max:255');
+
+
+            $group->getNodeByName('city')
+                ->label($block->getAttribute('cityLabel'))
+                ->placeholder($block->getAttribute('cityPlaceholder'))
+                ->rules('required', 'max:255');
+
+            $group->getNodeByName('state')
+                ->label($block->getAttribute('stateLabel'))
+                ->placeholder($block->getAttribute('statePlaceholder'))
+                ->rules('required', 'max:255');
```

Comment:
We don't have to do this now, but I want to add a reminder that we'll probably need to make use of the visibility conditions for `state` so it's hidden for countries that don't have states.

Reply:
I was taking a look at this and probably we'll need to make it respect not only the visibility according to the country, but also if it should be a text field or a select field. The v2 forms also load the states automatically on the front end when a select field is used. I recorded this video to demonstrate the different behaviors: https://www.loom.com/share/f530be8d1eca4ee682c7ee57e68203ce?sid=314a4195-2855-45a0-8dbe-19aee698f760

So, I believe that we need to handle it now because Ben already asked about it here:
https://www.loom.com/share/378a6f50a22e492babfb23371a0702da?highlightComment=12797555&t=66

Reply:
@glaubersilva interesting, yeah I guess we need to think about this one.  It also looks like GiveWP is loading all the state options on the server and looking for additional country/state settings saved.  I think we have a couple options for the new implementation:

1. Have the state field use the visibility conditions to register separate select & text fields with visibility and options based off the country (so only one works at at a time).  I'm not sure if this would work properly, but In theory this would mean we could accomplish everything with the fields api.
2. Make the state field a hidden field, and handle input changes and options on the client.

Reply:
Resolved with a bunch of commits between Jun 28, 2023 and Jul 1, 2023